File: Misc/fast_factor.py
def arrayF(n):
	F = [0] * (n+1)
	i = 2
	while (i*i <= n ):
		if(F[i] == 0 ):
			k = i*i
			while( k <= n):
				if( F[k] == 0):
					F[k] = i
				k += i
		i += 1
	return F



def factorization(x,F):
	primefactors = []
	while(F[x] > 0):
		primefactors += [F[x]]
		x = x // F[x]
	primefactors += [x]
	return  primefactors




if __name__ == "__main__":
	factors = arrayF(100)
	print(factorization(4,factors))

File: Misc/fast_factor.cpp
#include<iostream>
#include<vector>
typedef long long LL;
using namespace std;

vector<LL> Factors;

void createArray(LL N)
{
	for (LL i = 0; i < N+1; ++i)
		Factors.push_back(0);
	

	LL j = 2;

	while(j*j <= N)
	{
		if(Factors[j] == 0)
		{
			LL k = j * j ;
			while( k <= N )
			{
				if( Factors[k] == 0)
					Factors[k] = j;
				k += j;
			}
		}

		 j += 1;
	}

}

void FastFactorization(LL x)
{
	std::vector<LL> primefactors;

	while(Factors[x] > 0)
	{
		primefactors.push_back(Factors[x]);
		x /= Factors[x];
	}

	primefactors.push_back(x);


	for(int i = 0; i < primefactors.size() ;i++)
		cout<<primefactors[i]<<" ";
	cout<<endl;

}


int main()
{

	LL Number;
	cin>>Number;

	createArray(Number);

	LL factor;
	cin>>factor;

	
	FastFactorization(factor);


	return 0;
}
